src/scripts/main.js

```diff
@@ -1,3 +1,239 @@
 'use strict';
 
-// write your code here
+const startBtn = document.querySelector('.start');
+const fieldRows = document.querySelector('tbody').rows;
+const scoreBoard = document.querySelector('.game-score');
+
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+
+const rows = 4;
+const columns = 4;
+let score = 0;
+
+let field = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+startBtn.addEventListener('click', () => {
+  field = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  score = 0;
+
+  updateTheGame();
+  setNewCell();
+  setNewCell();
+
+  startBtn.classList.add('restart');
+  startBtn.innerText = 'Restart';
+
+  messageStart.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+});
+
+function setNewCell() {
+  if (!checkIfEmpty()) {
+    return;
+  }
+
+  const value = Math.random() > 0.1 ? 2 : 4;
+
+  let check = false;
+
+  while (!check) {
+    const row = Math.floor(Math.random() * rows);
+    const col = Math.floor(Math.random() * columns);
+
+    if (field[row][col] === 0) {
+      field[row][col] = value;
+      check = true;
+
+      updateTheGame();
+    }
+  }
+}
+
+function updateTheGame() {
+  for (let row = 0; row < rows; row++) {
+    for (let col = 0; col < columns; col++) {
+      const currentCell = fieldRows[row].cells[col];
+      const num = field[row][col];
+
+      updateCell(currentCell, num);
+    }
+  }
+
+  scoreBoard.innerText = String(score);
+}
+
+function updateCell(cell, num) {
+  cell.innerText = '';
+  cell.classList.value = '';
+  cell.classList.add('field-cell');
+
+  if (num > 0) {
+    cell.innerText = String(num);
+    cell.classList.add(`field-cell--${String(num)}`);
+  }
+}
+
+function checkIfEmpty() {
+  for (let row = 0; row < rows; row++) {
+    for (let col = 0; col < columns; col++) {
+      if (field[row][col] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+document.addEventListener('keyup', type => {
+  switch (startBtn.classList.contains('restart')) {
```

Comment:
`startBtn.classList.contains('restart')` - it should be inside `if`
```suggestion
  switch (event.code) {
```